src/Solvers/plan_transforms.jl

```diff
@@ -174,45 +168,49 @@ function plan_transforms(grid::VerticallyStretched, storage, planner_flag)
         # domains, or for domains that are `Bounded, Periodic, Bounded`.
         
         forward_periodic_plan = plan_forward_transform(storage, Periodic(), periodic_dims, planner_flag)
-        forward_bounded_plan = plan_forward_transform(storage, Bounded(), bounded_dims, planner_flag)
+        forward_bounded_plan  = plan_forward_transform(storage, Bounded(),  bounded_dims,  planner_flag)
 
-        forward_transforms = (DiscreteTransform(forward_bounded_plan, Forward(), grid, bounded_dims),
+        forward_transforms = (DiscreteTransform(forward_bounded_plan,  Forward(), grid, bounded_dims),
                               DiscreteTransform(forward_periodic_plan, Forward(), grid, periodic_dims))
 
         backward_periodic_plan = plan_backward_transform(storage, Periodic(), periodic_dims, planner_flag)
-        backward_bounded_plan = plan_backward_transform(storage, Bounded(), bounded_dims, planner_flag)
+        backward_bounded_plan  = plan_backward_transform(storage, Bounded(),  bounded_dims,  planner_flag)
 
         backward_transforms = (DiscreteTransform(backward_periodic_plan, Backward(), grid, periodic_dims),
-                               DiscreteTransform(backward_bounded_plan, Backward(), grid, bounded_dims))
+                               DiscreteTransform(backward_bounded_plan,  Backward(), grid, bounded_dims))
 
-    elseif !(Bounded in (TX, TY))
-        # We're on the GPU and either (Periodic, Periodic), (Flat, Periodic), or (Periodic, Flat) in xy.
-        # So, we pretend like we need a 2D doubly-periodic transform (even if one dimension is Flat).
-        forward_periodic_plan = plan_forward_transform(storage, Periodic(), [1, 2], planner_flag)
-        backward_periodic_plan = plan_backward_transform(storage, Periodic(), [1, 2], planner_flag)
+    elseif bounded_dims == ()
+        # We're on the GPU and either (Periodic, Periodic), (Flat, Periodic), or
+        # (Periodic, Flat) in the regular dimensions. So, we pretend like we need a 2D
+        # doubly-periodic transform (even if one dimension is Flat).
+        forward_periodic_plan = plan_forward_transform(storage, Periodic(), reg_dims, planner_flag)
+        backward_periodic_plan = plan_backward_transform(storage, Periodic(), reg_dims, planner_flag)
 
-        forward_transforms = tuple(DiscreteTransform(forward_periodic_plan, Forward(), grid, [1, 2]))
-        backward_transforms = tuple(DiscreteTransform(backward_periodic_plan, Backward(), grid, [1, 2]))
+        forward_transforms = tuple(DiscreteTransform(forward_periodic_plan, Forward(), grid, reg_dims))
```

Comment:
can you batch here if you have reg_dims = [1, 3]? probably you cannot and have to handle this case as the one below

Reply:
So far this seems to be working and the only problems seem to happen with the case below. But I'll test more thoroughly